### PR TITLE
Support isolated clusters

### PR DESCRIPTION
**Pull request by majst01**
Related to: https://github.com/fi-ts/proxy-services/issues/4

### CODE REVIEW

File: cmd/cluster.go [hunk 1]
```diff
@@ -487,6 +495,39 @@ func (c *config) clusterCreate() error {
...
+	var networkAccessType *string
+	if viper.IsSet("network-isolation") {
+		networkAccessType = pointer.Pointer(viper.GetString("network-isolation"))
+		switch *networkAccessType {
+		case models.V1ClusterCreateRequestNetworkAccessTypeForbidden:
+			fmt.Printf(`
+WARNING: You are going to create a cluster which has no internet access with the following consequences:
+- pulling images is only possible from private registries you provide, these registries must be resolvable from the public dns and must be secured with a trusted TLS certificate
+- service type loadbalancer can only be created in networks which are not internet facing
+- cluster wide network policies can only be create in networks which are not internet facing
```
Comment (Gerrit91) [inline cmd/cluster.go:507]:
```suggestion
- cluster wide network policies can only be created in networks which are not internet facing
```

File: cmd/cluster.go [hunk 2]
```diff
@@ -487,6 +495,39 @@ func (c *config) clusterCreate() error {
...
+			err := helper.Prompt("Are you sure? (y/n)", "y")
+			if err != nil {
+				return err
+			}
+		case models.V1ClusterCreateRequestNetworkAccessTypeRestricted:
+			fmt.Printf(`
+WARNING: You are going to create a cluster which has no internet access with the following consequences:
+- pulling images is only possible from private registries you provide, these registries must be resolvable from the public dns and must be secured with a trusted TLS certificate
+- you can create cluster wide network policies to the outside world without restrictions
+- but cluster wide network policies allowing access to popular registries like docker.io, quay.io and eu.gcr.io will still not allow pulling images from there
```
Comment (mwennrich) [inline cmd/cluster.go:519]: not true, with 0.0.0.0/0 port 443 images can also be pulled from these registries
```suggestion
```

File: cmd/cluster.go [hunk 3]
```diff
@@ -487,6 +495,39 @@ func (c *config) clusterCreate() error {
...
 	}
 
+	var networkAccessType *string
+	if viper.IsSet("network-isolation") {
+		networkAccessType = pointer.Pointer(viper.GetString("network-isolation"))
+		switch *networkAccessType {
+		case models.V1ClusterCreateRequestNetworkAccessTypeForbidden:
+			fmt.Printf(`
+WARNING: You are going to create a cluster which has no internet access with the following consequences:
+- pulling images is only possible from private registries you provide, these registries must be resolvable from the public dns and must be secured with a trusted TLS certificate
```
Comment (mwennrich) [inline cmd/cluster.go:505]:
```suggestion
- pulling images is only possible from private registries you provide, these registries must be resolvable from the public dns, their IP must be located in one of the allowed networks (see cluster inputs), and must be secured with a trusted TLS certificate
```

File: cmd/cluster.go [hunk 4]
```diff
@@ -487,6 +495,39 @@ func (c *config) clusterCreate() error {
...
+- service type loadbalancer can only be created in networks which are specified in the allowed networks (see cluster inputs)
+- cluster wide network policies can only be created in certain network ranges whichspecified in the allowed networks (see cluster inputs)
```
Comment (mwennrich) [inline cmd/cluster.go:507]:
```suggestion
- cluster wide network policies can only be created in certain network ranges specified in the allowed networks (see cluster inputs)
```

File: cmd/cluster.go [hunk 5]
```diff
@@ -487,6 +495,39 @@ func (c *config) clusterCreate() error {
...
+- It is not possible to change this cluster back to %q after creation
+`, models.V1ClusterCreateRequestNetworkAccessTypeBaseline)
+			err := helper.Prompt("Are you sure? (y/n)", "y")
+			if err != nil {
+				return err
+			}
+		case models.V1ClusterCreateRequestNetworkAccessTypeRestricted:
+			fmt.Printf(`
+WARNING: You are going to create a cluster which has no internet access with the following consequences:
```
Comment (mwennrich) [inline cmd/cluster.go:516]:
```suggestion
WARNING: You are going to create a cluster that has no default internet access with the following consequences:
```

File: cmd/cluster.go (hunk 2 above)
Comment (majst01) [reply cmd/cluster.go]: already adopted this sentence

**Review by mwennrich — APPROVED** [review]

**Pull request merged**